lib/swaps/Swaps.ts

```diff
@@ -76,11 +79,13 @@ interface Swaps {
 class Swaps extends EventEmitter {
   /** A map between r_hash and swap deals. */
   private deals = new Map<string, SwapDeal>();
+  private hashs = new Set<string>();
+  private repository: SwapRepository;
 
-  constructor(private logger: Logger, private pool: Pool, private lndBtcClient: LndClient, private lndLtcClient: LndClient) {
+  constructor(private logger: Logger, private models: Models, private pool: Pool, private lndBtcClient: LndClient, private lndLtcClient: LndClient) {
     super();
+    this.repository = new SwapRepository(this.models);
 
-    this.bind();
```

Comment:
I'd leave this in the constructor, that's how it's done in other classes with a similar bind function.